rust/lance/src/dataset/write.rs

```diff
@@ -72,6 +77,7 @@ impl Default for WriteParams {
             mode: WriteMode::Create,
             store_params: None,
             progress: None,
+            tail_compaction_size: None,
```

Comment:
I would set the default to `max_rows_per_group` or just 1024. Our benchmarking showed that it's not the # of fragments that leads to slower queries as much as number of groups. So compacting until we have a full group would be wise.